src/CoreBundle/private/library/classes/dbobjects/TCMSUser.class.php

```diff
@@ -412,37 +268,18 @@ public function GetCurrentEditLanguageObject()
      *
      * @throws ErrorException
      * @throws TPkgCmsException_Log
+     * @deprecated 7.2 use \ChameleonSystem\CmsBackendBundle\BackendSession\BackendSession
      */
     public function SetCurrentEditLanguage($language = null)
     {
+        /** @var BackendSessionInterface $sessionService */
+        $sessionService = ServiceLocator::get('chameleon_system_cms_backend.backend_session');
         if (null === $language) {
-            $config = TdbCmsConfig::GetInstance();
-            $language = $config->GetFieldTranslationBaseLanguage()->fieldIso6391;
+            $sessionService->resetCurrentEditLanguage();
+            return;
         }
-        $databaseConnection = $this->getDatabaseConnection();
-        $updateQuery = 'UPDATE `cms_user` SET `cms_current_edit_language` = :language WHERE `id` = :userId';
-        $databaseConnection->executeQuery($updateQuery, array(
-            'language' => $language,
-            'userId' => $this->sqlData['id'],
-        ));
-        $this->sqlData['cms_current_edit_language'] = $language;
-
-        // reset language object cache
-        $this->SetInternalCache('oCurrentEditLanguage', null);
-
-        $_SESSION['cmsbackend-currenteditlanguage'] = $language;
-        $this->GetCurrentEditLanguage(true);
-    }
 
-    /**
-     * Loads the access manager for the user (controls access to tables and modules).
-     */
-    public function _LoadAccessManager()
-    {
-        if (!is_null($this->id)) {
-            $this->oAccessManager = new TAccessManager();
-            $this->oAccessManager->InitFromObject($this);
-        }
+        $sessionService->setCurrentEditLanguageIso6391($language);
     }
 
     /**
```

Comment:
That's a lot of public stuff vanishing. I understand why, but at this level we need to drop the idea that this could be version 7.2. This clearly needs to be version 8.0

Reply:
jup. It will be 8.0